The job **`x86_64-gnu-aux`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/14678045193/job/41197173391) [(plain)](https://github.com/rust-lang-ci/rust/commit/742814d5c2c47a2797ff79bcdbd2121010fb2c35/checks/41197173391/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config submodule.library/backtrace.active=true
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
file:.git/config submodule.library/stdarch.url=https://github.com/rust-lang/stdarch.git
file:.git/config submodule.src/doc/book.active=true
---
test diesel/src/upsert/on_conflict_extension.rs - upsert::on_conflict_extension::InsertStatement<T,U,Op,Ret>::on_conflict (line 96) ... ok

failures:

---- diesel/src/expression/functions/aggregate_folding.rs - expression::functions::aggregate_folding::avg (line 30) stdout ----
error: expected one of `async`, `|`, or `||`, found `diesel`
##[error] --> diesel/src/expression/functions/../../doctest_setup.rs:1:5
  |
1 | use diesel::prelude::*;
  |     ^^^^^^ expected one of `async`, `|`, or `||`

error: non-statement macro in statement position: include
##[error] --> diesel/src/expression/functions/aggregate_folding.rs:35:1
  |
8 | include!("../../doctest_setup.rs");
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot find macro `table` in this scope
##[error]  --> diesel/src/expression/functions/aggregate_folding.rs:42:1
   |
---

thread 'main' panicked at src/tools/cargotest/main.rs:123:9:
tests failed for https://github.com/diesel-rs/diesel
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:18:05
make: *** [Makefile:51: check-aux] Error 1
  local time: Sat Apr 26 05:37:06 UTC 2025
  network time: Sat, 26 Apr 2025 05:37:06 GMT
##[error]Process completed with exit code 2.
Post job cleanup.
```

</details>

--- a/src/librustdoc/doctest/make.rs
+++ b/src/librustdoc/doctest/make.rs
@@ -424,7 +424,10 @@ fn parse_source(source: &str, crate_name: &Option<&str>) -> Result<ParseSourceIn
                     // We assume that the macro calls will expand to item(s) even though they could
                     // expand to statements and expressions. And the simple fact that we're trying
                     // to retrieve a `main` function inside it is a terrible idea.
-                    StmtKind::MacCall(ref mac_call) if !info.has_main_fn => {
+                    StmtKind::MacCall(ref mac_call) => {
+                        if info.has_main_fn {
+                            continue;
+                        }
                         let mut iter = mac_call.mac.args.tokens.iter();
 
                         while let Some(token) = iter.next() {

--- a/tests/rustdoc-ui/doctest/auxiliary/macro-after-main.rs
+++ b/tests/rustdoc-ui/doctest/auxiliary/macro-after-main.rs
@@ -1,0 +1,1 @@
+use std::string::String;

--- a/tests/rustdoc-ui/doctest/macro-after-main.rs
+++ b/tests/rustdoc-ui/doctest/macro-after-main.rs
@@ -1,0 +1,16 @@
+// This test checks a corner case where the macro calls used to be skipped,
+// making them considered as statement, and therefore some cases where
+// `include!` macro was then put into a function body, making the doctest
+// compilation fail.
+
+//@ compile-flags:--test
+//@ normalize-stdout: "tests/rustdoc-ui/doctest" -> "$$DIR"
+//@ normalize-stdout: "finished in \d+\.\d+s" -> "finished in $$TIME"
+//@ check-pass
+
+//! ```
+//! include!("./auxiliary/macro-after-main.rs");
+//!
+//! fn main() {}
+//! eprintln!();
+//! ```

--- a/tests/rustdoc-ui/doctest/macro-after-main.stdout
+++ b/tests/rustdoc-ui/doctest/macro-after-main.stdout
@@ -1,0 +1,6 @@
+
+running 1 test
+test $DIR/macro-after-main.rs - (line 11) ... ok
+
+test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in $TIME
+

